### PR TITLE
Natural Gas introduction

### PR DESCRIPTION
**Pull request by N3rdP1um23**
As mentioned in https://github.com/jrfernandes/ontario_energy_board/issues/13, the following PR includes the additional of Natural Gas prices!! 🎊 

Since the last main version
- There only one **major breaking change**
  - Users will have to select their electricity companies again as the `entity id` has changed to include the energy sector
- There's only a single *potential breaking change*
  - Some of the newer keys (https://github.com/jrfernandes/ontario_energy_board/commit/0bf7fa7a7d2ed7b1e95358ffedcc954a54d6a152) have been changed slightly to follow the definitions from OEB - which may require users to modify existing calls for attritional attributes

The flow still remains the same when selecting a Natural Gas option. When you go to add a new OEB integration, all energy (electricity & natural gas) companies/options will be listed. The company names (and furthermore the `entity id`) has changed slightly
- **Old:** `COMPANY (CLASS)`
- **New:** `COMPANY (CLASS) [SECTOR]`

Other than that, I just need to clean up some of the formulas that I've received from OEB to calculate the potential monthly bill. Once that's done, I'll share an update as well!

If you have any questions/comments/concerns, feel free to let me know! 😃 

**PS** Included in this update, I've also taken care of the following issue (https://github.com/jrfernandes/ontario_energy_board/issues/21) where a deprecated function call was being used.

### CODE REVIEW

**Comment by N3rdP1um23** [comment]
Hey! Just wanted to follow up and see if there was any update on this PR? :) Let me know if you need anything!

**Comment by jrfernandes** [comment]
Thank you for your contribution, @N3rdP1um23. I'll try my best to do a code review during the weekend.

**Comment by N3rdP1um23** [comment]
Hey @jrfernandes! My pleasure! Happy to contribute and future the extension for HA. Oh no worries at all and take your time 🙂

**Comment by jrfernandes** [comment]
@N3rdP1um23 - There are some merge conflicts to be solved. Also, I've added [black](https://github.com/psf/black) verification for consistent styling. You might have to update part of the code here for tests to pass (once `main` is merged here).

**Comment by N3rdP1um23** [comment]
@jrfernandes Hey! Thanks for the update and enhancements to the repo. I've just gone through and have resolved the merge conflicts, upgraded using Black (and also formatted this PR), as well as resolving any lint issues. We should be good to go now :)

Let me know if you need anything else!

**Comment by N3rdP1um23** [comment]
Just bumped up the all the minimum dependency versions to the latest version.

**Comment by Digital-Ark** [comment]
In case this helps anyone, this is how I'm reading my natural gas meter with a digital compass sensor.

I am eagerly awaiting this update that scrapes the OEB gas rates.

[https://community.home-assistant.io/t/attempting-to-read-an-elster-acm-250-dumb-gas-meter-with-a-qmc5883l/600899](url)

**Comment by nromyn** [comment]
Bumping @jrfernandes i would really appreciate this capability.

**Review by jrfernandes — APPROVED** [review]
Let's get this out. I've been hesitant as I was hoping to get some cleanup/refactoring in a few places, but time hasn't been on my side.

As-is, this is a great addition and makes it better. We should not delay it any further.

Let's :shipit:

File: custom_components/ontario_energy_board/manifest.json [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-    "hijri_converter>=2.2.4"
-  ],
-  "codeowners": [
-    "@jrfernandes"
-  ],
-  "iot_class": "cloud_polling",
-  "version": "0.3.6"
+	"domain": "ontario_energy_board",
+	"name": "Ontario Energy Board",
+	"codeowners": ["@jrfernandes", "@n3rdp1um23"],
```
Comment (jrfernandes) [inline custom_components/ontario_energy_board/manifest.json:4]: With great power comes great responsibility :)

File: custom_components/ontario_energy_board/manifest.json [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+	"config_flow": true,
+	"documentation": "https://github.com/jrfernandes/ontario_energy_board",
+	"iot_class": "cloud_polling",
+	"issue_tracker": "https://github.com/jrfernandes/ontario_energy_board/issues",
+	"requirements": [
+		"beautifulsoup4==4.12.3",
+		"holidays>=0.46",
+		"hijri_converter>=2.3.1"
+	],
+	"version": "0.4.0"
```
Comment (jrfernandes) [inline custom_components/ontario_energy_board/manifest.json:14]: 💪 💪 💪

**Comment by jrfernandes** [comment]
That's great to hear. I appreciate the help.

Thank you.

**Pull request merged**